supabase/models/updateRequestToReserve.ts

```diff
@@ -0,0 +1,39 @@
+import newClient from '../utils/newClient';
+
+export const updateRequestToReserve = async (
+  itemId: number,
+  userId: string
+) => {
+  try {
+    const supabase = newClient();
+
+    const { data: item, error: fetchError } = await supabase
+      .from('items')
+      .select('requestedToReserve')
+      .eq('id', itemId)
+      .single();
+
+    if (fetchError) {
+      throw new Error(fetchError.message);
+    }
+
+    const currentRequestedToReserve = item?.requestedToReserve || [];
+    if (currentRequestedToReserve.includes(userId)) {
+      alert('You have already requested this item.');
+    } else {
+      const updatedRequestedToReserve = [...currentRequestedToReserve, userId];
```

Comment:
```suggestion
    if (data.requestedToReserve.includes(userId)) {
      alert('You have already requested this item.');
    } 
    
    const updatedRequestedToReserve = [...data.requestedToReserve, userId];
```

Don't hide a bad database return behind a default value - just return what's there and if it doesn't match the types we expect we should be handling that - that's why we have TypeScript!

We also just don't need that `else`, we can remove some indentation and make this much more readable. 

Similarly, we only really need to create a copy of our data if we are going to do any manipulation, otherwise we're not risking it's immutability anyway. So we can just create a new copy if we want to add a new user but we don't need it just in order to check if it includes the current user already

Reply:
Keeping the if/else structure allows us to clearly distinguish between different scenarios:
1. **User Already Requested**: We explicitly check if the user has already made a request and return an appropriate message. This ensures users are informed of the duplicate request status.
2. **Update Request**: If the user has not made a request, we proceed to update the reservation list. This separation maintains clear logical flow and error handling.

This approach helps in debugging and ensures that each condition is handled explicitly, making the code more maintainable and easier to understand.

Reply:
-  Removed unnecessary else blocks to improve readability.
-  Retained `const currentRequestedToReserve = item?.requestedToReserve || [];` to handle cases where `requestedToReserve` may be `null`(in our case the initial value the column(requestToReserve) is null), preventing data errors and ensuring the reservation request process functions correctly.